app/services/SessionService.scala

```diff
@@ -100,6 +100,31 @@ class SessionService @Inject()(
       uiJourneySessionDataRepository.set(uiJourneySessionData)
   }
 
+  def setMultipleMongoData(kv: Map[String, String], journeyType: JourneyType)
+                 (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[Throwable, Boolean]] = {
+    val uiJourneySessionData = UIJourneySessionData(hc.sessionId.get.value, journeyType.toString)
+    kv.map { kv_ =>
+        journeyType.operation match {
+          case Add => (kv_._2, AddIncomeSourceData.getJSONKeyPath(kv_._1))
+          case Manage => (kv_._2, ManageIncomeSourceData.getJSONKeyPath(kv_._1))
+          case Cease => (kv_._2, CeaseIncomeSourceData.getJSONKeyPath(kv_._1))
+        }
+      }.toList
+      .foldLeft(Future.successful[Either[Throwable, Boolean]](Right(false))) { (acc, kv) =>
+        acc.flatMap {
+          case Right(_) =>
+            uiJourneySessionDataRepository.updateData(uiJourneySessionData, kv._2, kv._1).map(
+              result => result.wasAcknowledged() match {
+                case true => Right(true)
+                case false => Left(new Exception("Mongo Save data operation was not acknowledged"))
+              }
+            )
+          case Left(ex) =>
+            Future.successful(Left(ex))
+        }
+      }
+  }
+
   def setMongoKey(key: String, value: String, journeyType: JourneyType)
```

Comment:
this method will become a special case of setMultipleMongoData, as you can wire it to use setMultipleMongoData;